play-media-native/features/ImageGrid/ImageOverlayCarousel.tsx

```diff
@@ -0,0 +1,60 @@
+import { Dimensions, Image, StyleSheet } from "react-native";
+import Carousel from "react-native-reanimated-carousel";
+import { IconButton, Modal, Portal } from "react-native-paper";
+
+export const ImageOverlayCarousel = ({
+  images,
+  startIndex = 0,
+  visible,
+  onClose,
+}: {
+  images: string[];
+  startIndex?: number;
+  visible: boolean;
+  onClose: () => void;
+}) => {
+  const width = Dimensions.get("window").width;
+  const height = Dimensions.get("window").height;
+
+  const styles = StyleSheet.create({
+    visibilityStyle: {
+      display: visible ? "flex" : "none",
+    },
+  });
+
+  return (
+    <Portal>
+      <Modal visible={visible} onDismiss={onClose}>
+        <Carousel
+          loop
+          width={width}
+          height={height}
+          data={images}
+          defaultIndex={startIndex}
+          renderItem={({ item }) => {
+            return (
+              <Image
+                source={{ uri: item }}
+                style={{
+                  height: height,
+                  width: width,
+                  resizeMode: "contain",
+                }}
+              />
+            );
+          }}
+        />
+        <IconButton
```

Comment:
Another minor thing, the close button is not very visible on some event detail pages 

![image](https://user-images.githubusercontent.com/96472119/214001606-df6aafcf-60d3-4882-861f-3b830caa1884.png)